### ISSUE TITLE
GraphQL Live Preview: New entries without saving are not returned 

### ISSUE THREAD

**Issue by carola-kolossal** [statamic/cms#6134]
### Bug description

Cant't use Live Preview as expected see [https://statamic.dev/live-preview#headless-frontend-frameworks](https://statamic.dev/live-preview#headless-frontend-frameworks). Only saved entries are returned by GraphQL. New Entries without saving are missing.

### How to reproduce

1. Open Statamic CMS
2. Create new entry without saving it
3. Open 'Live Preview'
4. Pass corresponding token as parameter to GraphQL and try to retrieve new entry via GraphQL

Result: Entry is not returned by GraphQL

5. Save entry
Result: Entry is returned by GraphQL

### Versions

3.3.7

### Installation

Fresh statamic/statamic site via CLI

**Comment by luigi-finaldream** [comment]
![image](https://user-images.githubusercontent.com/100590991/176026580-ee74de6f-c995-43a6-a7ee-40875eeac624.png)

**Comment by luigifgp** [comment]
![image](https://user-images.githubusercontent.com/60223731/176026726-67b81941-6d60-4df3-9881-6fb699fdd0dc.png)

**Comment by jasonvarga** [comment]
@luigifgp This issue is about a brand new entry, e.g. one that is still in progress on the "create" form. It hasn't been saved yet. Has yours been saved?

**Comment by luigifgp** [comment]
@jasonvarga yes i'm saving it as a draft but the problem persists the only way i get value is by posting it but i guess that shouldn't be a good idea if i haven't finished my post and my frontend syncs with a broken post

**Comment by edalzell** [comment]
We are using "normal" live preview and getting the same thing

**Comment by danielbrodin** [comment]
I have the same issue.
- New unsaved entries returns nothing
- Entries in draft returns nothing
- Published entries works like expected

**Comment by carola-kolossal** [comment]
> I'm having the same issue it only returns values when I save and published, not when I save as a draft, @carola-kolossal do you have solved something about this ?

@luigifgp The issue only exist if an entry is **new** and **not saved**. Unpublished and saved entries are returned by GraphQL as expected. Set the filter to `filter: {status: "draft"}` or if you like to get both use regex `filter: {status: {regex: "(draft|published)"}}`. If nothing is set only 'published' entries are returned.

<img width="911" alt="Screenshot 2022-07-05 at 13 16 38" src="https://user-images.githubusercontent.com/72194401/177315848-147f73dc-08b3-452a-b92b-05329a56c665.png">

**Comment by el-schneider** [comment]
While filtering for `(draft|published)` works, including drafts in live-preview should be the default behaviour.

**Comment by sc-urs** [comment]
Hello,

This github issue is the closest thing I could find that would describe the problem I'm having.

Basically, I have a Next.JS frontend headlessly connected to Statamic CMS via GraphQL.

The live preview function is connected properly and it works when editing already created entries.

The problem arises on the create entry page, where the live preview button is available but the post has not yet been saved nor published.

There is no slug to which the live preview can point to in this case, because it has not yet been saved inside entries for graphQL to read, therefore the content I enter inside the entry has no way to show in the live preview.

What I'm wondering is if this is a feature that is not implemented yet or am I missing something here?

Thank you for your time!

**Comment by jasonvarga** [comment]
> This github issue is the closest thing I could find that would describe the problem I'm having.

It's literally the same problem. You've come to the right place!

**Comment by jakubjo** [comment]
I also have the same problem trying to access the not yet saved entry in PHP as recommended in the docs: https://statamic.dev/live-preview#custom-rendering

**Comment by CptCompiler** [comment]
any updates on this one?

**Comment by Smef** [comment]
I tried using the `filter: {status: "draft"}` option but this is returning a `Forbidden: status` error. This includes sending with the token query parameter. Is there another thing which needs to be configured to be able to see these entries?

**Comment by jasonvarga** [comment]
> I tried using the filter: {status: "draft"} option but this is returning a Forbidden: status error. 

Filters are disabled by default. You need to opt into it. https://statamic.dev/graphql#filtering

**Comment by Smef** [comment]
Ah, yes. I had that enabled for the REST API, but not GraphQL. Thanks!

This does allow it to be queried, but unfortunately it also exposes drafts to the public, which isn't great.

**Comment by TheBnl** [comment]
In `Statamic\GraphQL\QueriesEntriesQuery` on line 68 there is this block of code that filters the draft entries even if a token is set. 
```php
private function filterQuery($query, $filters)
    {
        if (! isset($filters['status']) && ! isset($filters['published'])) {
            $filters['status'] = 'published';
        }

        $this->traitFilterQuery($query, $filters);
    }
```

Changing it to the following causes the draft entries to be included in the output:
```
private function filterQuery($query, $filters)
    {
        if (!request()->isLivePreview() && (! isset($filters['status']) && ! isset($filters['published']))) {
            $filters['status'] = 'published';
        }

        $this->traitFilterQuery($query, $filters);
    }
``` 

I can make a PR to fix this but i'm unsure on how many of the Queries would need to be altered. I can start with the Entries and Entry queries if you'd like :+1:
